src/overlays/actors/ovl_En_Zog/z_en_zog.c

```diff
@@ -43,78 +48,1047 @@ const ActorInit En_Zog_InitVars = {
+static AnimationHeader* D_80B95920 = 0x6002344;
+static AnimationHeader* D_80B95924 = 0x6002894;
+static AnimationHeader* D_80B95928 = 0x60030E0;
+static AnimationHeader* D_80B9592C = 0x60037F8;
+static AnimationHeader* D_80B95930 = 0x60041D0;
+static AnimationHeader* D_80B95934 = 0x6004BDC;
+static AnimationHeader* D_80B95938 = 0x60055B4;
+
+static s16 D_80B9593C[] = {
+    0x0000,
+    0x0001,
+    0x0002,
+    0x0003,
+};
+
+static s16 D_80B95944[] = {
+    4,
+    5,
+};
+
+static s16 D_80B95948[] = {
+    0x0005,
+    0x0006,
+    0x0007,
+    0x0000,
+};
+
+static s16 D_80B95950[] = {
+    0x0007, 0x0008, 0x0009, 0x000A, 0x0007, 0x000B, 0x000C, 0x000D, 0x000E,
+    0x000F, 0x000C, 0x000D, 0x000E, 0x000F, 0x000C, 0x0010, 0x0011, 0x0000,
+};
+
+static s16* D_80B95974[] = {
+    D_80B9593C,
+    D_80B95944,
+    D_80B95948,
+    D_80B95950,
+};
+
+static s16 D_80B95984[] = {
+    0x0004,
+    0x0002,
+    0x0003,
+    0x0011,
+};
+
+static Vec3f D_80B9598C = { 0.0f, -0.05f, 0.0f };
+static Vec3f D_80B95998 = { 0.0f, -0.025f, 0.0f };
+static Color_RGBA8 D_80B959A4 = { 220, 220, 255, 255 };
+static Color_RGBA8 D_80B959A8 = { 80, 80, 220, 255 };
+static Vec3f D_80B959AC = { 0.0f, 0.0f, 15.0f };
+
+void func_80B93310(Actor* thisx, Lights* mapper, GlobalContext* globalCtx) {
+    Vec3f sp34;
+    EnZog* this = THIS;
+    f32 sp2C;
+
+    if (this->unk_322 > 0) {
+        if (this->unk_30A & 8) {
+            this->actor.shape.shadowAlpha = this->unk_322;
+        } else {
+            this->actor.shape.shadowAlpha = 255;
+        }
+
+        Math_Vec3f_Copy(&sp34, &this->actor.world.pos);
+
+        sp2C = sqrtf(SQ(this->actor.focus.pos.x - this->unk_2F0.x) + SQ(this->actor.focus.pos.z - this->unk_2F0.z));
+        if (sp2C < 12.0f) {
+            sp2C = 12.0f;
+        } else if (sp2C > 60.0f) {
+            sp2C = 60.0f;
+        }
+
+        Math_Vec3f_Copy(&this->actor.world.pos, &this->unk_2F0);
+        func_800B4AEC(globalCtx, &this->actor, 50.0f);
+        if (sp34.y < this->actor.floorHeight) {
+            this->actor.world.pos.y = this->actor.floorHeight;
+        } else {
+            this->actor.world.pos.y = sp34.y;
+        }
+        this->actor.scale.z *= sp2C * (1 / 12.0f);
+        func_800B3FC0(&this->actor, mapper, globalCtx);
+        this->actor.scale.z = this->actor.scale.x;
+        Math_Vec3f_Copy(&this->actor.world.pos, &sp34);
+    }
+}
+
+void func_80B93468(EnZog* this, GlobalContext* globalCtx) {
+    Vec3s* points;
+
+    this->unk_2E8 = &globalCtx->setupPathList[ENZOG_GET_FC00(&this->actor)];
+    if (this->unk_2E8 != NULL) {
+        points = &((Vec3s*)Lib_SegmentedToVirtual(this->unk_2E8->points))[this->unk_2EC];
+        points++;
+
+        this->actor.world.pos.x = points[-1].x;
+        this->actor.world.pos.z = points[-1].z;
+        this->actor.world.rot.y =
+            Math_Atan2S(points[0].x - this->actor.world.pos.x, points[0].z - this->actor.world.pos.z);
```

Comment:
I personally think using `->` instead of `[0]` here makes a bit more sense
```suggestion
            Math_Atan2S(points->x - this->actor.world.pos.x, points->z - this->actor.world.pos.z);
```

Reply:
It's using [-1] right above, I think it fits better to use [0] and [-1] rather than -> and [-1]. I changed it though.